doc/user/layouts/partials/create-source/connector/s3/details.html

```diff
@@ -0,0 +1,44 @@
+### S3 source details
+
+The S3 source is designed to ingest a large volume of static data from S3.
+
+#### S3 Limitations
+
+Currently S3 sources do not support Avro-encoded objects.
+
+All S3 sources are append-only -- deleted and updated [S3 Objects][] are silently ignored.
+
+[S3 Objects]: https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingObjects.html
+
+#### Scanning S3 Buckets
+
+The **OBJECTS FROM SCAN** syntax causes S3 sources to issue [ListBucket][] requests to collect the
+names of objects to collect. After collecting the list of keys, their names are checked against
+the **MATCHING** clause, if present, before downloading and extracting objects line by line.
+
+[ListBucket]: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html
+
+#### Patterns
+
+It is possible to filter the list of object keys to download using unix-style glob syntax as an
+argument to the `MATCHING` clause:
+
+* `?` matches any single character except `/`.
+* `*` matches zero or more characters except `/`.
+* `**` recursively matches directories, but some other pattern must be specified. For example
+`a/**` matches anything inside of the `a/` prefix (but not `a/` itself), and `**/a` matches `a`
+in any prefix, but not `a` with no prefix.
+* `{a,b}` matches a or b where a and b are arbitrary glob patterns.
+* `[ab]` matches `a` or `b` where a and b are characters. Use `[!ab]` to match any character
```

Comment:
nit: rather than providing a `!` example, explaining it is a little clearer, e.g.

> Prepending `!` to the matched characters inverts the match, e.g. `[!ab]` matches any character besides `a` or `b`.